src/macaron/__main__.py

```diff
@@ -92,15 +93,22 @@ def analyze_slsa_levels_single(analyzer_single_args: argparse.Namespace) -> None
         branch = analyzer_single_args.branch
         digest = analyzer_single_args.digest
 
-        if repo_path and purl and not digest:
-            # To provide the purl together with the repository path, the user must specify the branch and commit
-            # digest.
-            logger.error(
-                "Please provide the commit digest for the repo at %s that matches to the PURL string %s.",
-                repo_path,
-                purl,
-            )
-            sys.exit(os.EX_USAGE)
+        if repo_path and purl:
+            # To provide the purl together with the repository path, the user must specify the commit digest unless the
+            # purl has a version.
+            try:
+                purl_object = PackageURL.from_string(purl)
+            except ValueError as error:
+                logger.debug("Could not parse PURL: %s", error)
+                purl_object = None
+            if not purl_object or not (purl_object.version or digest):
```

Comment:
Instead of setting `purl_object` to `None` and rechecking it in the later `if`, why don't we exit instead?
```suggestion
                sys.exit(os.EX_USAGE)
            if not (purl_object.version or digest):
```
The error message logged inside the following `if` statement does not mention anything about the PURL being invalid, but instead tells the user to provide the commit/version. If the PURL is invalid, this message would be irrelevant since it is pointing to a completely different issue.

Reply:
Agreed that exiting immediately if the PURL string is incorrect with a relevant error message would make sense. Then you can keep the following error for the case where neither the version nor the commit hash is provided.